src/BuildingRegistry.Api.Legacy/Infrastructure/Grb/Wfs/GrbWfsClient.cs

```diff
@@ -0,0 +1,134 @@
+namespace BuildingRegistry.Api.Legacy.Infrastructure.Grb.Wfs
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Xml;
+    using System.Xml.Linq;
+    using System.Xml.XPath;
+    using NetTopologySuite.Geometries;
+    using NetTopologySuite.IO.GML2;
+
+    public interface IGrbWfsClient
+    {
+        IEnumerable<Tuple<Geometry, IReadOnlyDictionary<string, string>>> GetFeaturesInBoundingBox(GrbFeatureType featureType, Envelope boundingBox);
+    }
+
+    internal class GrbWfsClient : IGrbWfsClient
+    {
+        private readonly string _wfsUrl;
+        private readonly XmlNamespaceManager _namespaceManager;
+        private readonly GMLReader _gmlReader;
+
+        public GrbWfsClient(GrbWfsConfiguration configuration)
+        {
+            if (configuration == null)
+                throw new ArgumentNullException(nameof(configuration));
+
+            _wfsUrl = configuration.Url;
+
+            _namespaceManager = new XmlNamespaceManager(new NameTable());
+            _namespaceManager.AddNamespace("wfs", GrbWfsNameSpaces.Wfs);
+            _namespaceManager.AddNamespace("gml", GrbWfsNameSpaces.Gml);
+            _namespaceManager.AddNamespace("grb", GrbWfsNameSpaces.Grb);
+
+            _gmlReader = new GMLReader();
+        }
+
+        public IEnumerable<Tuple<Geometry, IReadOnlyDictionary<string, string>>> GetFeaturesInBoundingBox(GrbFeatureType type, Envelope boundingBox)
+        {
+            var featureName = GrbFeatureName.For(type);
+            var wfsRequest = CreateWfsRequest(featureName, boundingBox);
+
+            try
+            {
+                var response = wfsRequest
+                        .GetResponse()
+                        .GetResponseStream();
```

Comment:
What if this throws an exception? No catch all in the catch for any kind of exception, wrapping it in a `GrbWfsException`, then we at least know the source was the Grb part.